### PR TITLE
Feature/hw 58111 java remove filters

### PR DESCRIPTION
**Pull request by jkurra-hw**

### CODE REVIEW

File: src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java [hunk 1]
```diff
@@ -69,7 +69,9 @@ public HyperwalletApiClient(final String username, final String password, final
     public <T> T get(final String url, final TypeReference<T> type) {
         Response response = null;
         try {
-            response = getService(url, true).getResource();
+            Request req = getService(url, true);
+            response = req.getResource();
+            //response = getService(url, true).getResource();
```
Comment (ramahalingam) [inline src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java:74]: Revert the changes in this file.

File: src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java [hunk 2]
```diff
@@ -154,8 +154,27 @@ public HyperwalletUser updateUser(HyperwalletUser user) {
      * @param options List filter option
      * @return HyperwalletList of HyperwalletUser
      */
-    public HyperwalletList<HyperwalletUser> listUsers(HyperwalletPaginationOptions options) {
+    public HyperwalletList<HyperwalletUser> listUsers(HyperwalletUsersListPaginationOptions options) {
         String url = paginate(this.url + "/users", options);
+        if (options != null) {
+            if (options.getClientUserId() != null) {
```
Comment (ramahalingam) [inline src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java:160]: The addParameter function already checks for null values. So just check whether the options is null or not and then call the addParameter without null checks. 
This is a generic comment. Please change in all the places.

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java [hunk 3]
```diff
@@ -487,14 +497,100 @@ public void testListBankCard() throws Exception {
...
+        HyperwalletLink actualHyperwalletLink = returnValue.getLinks().get(0);
+        HyperwalletLink expectedHyperwalletLink = hyperwalletLinks.get(0);
+        assertThat(actualHyperwalletLink.getHref(), is(equalTo(expectedHyperwalletLink.getHref())));
+        assertEquals(actualHyperwalletLink.getParams(), expectedHyperwalletLink.getParams());
+    }
+
+    /*
+    Following test is not returning response. This is dissabled timebeing. To be rerun after enabling.
+     */
+    @Test(enabled = false)
```
Comment (ramahalingam) [inline src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java:509]: Please enable all the tests.

**Review by ramahalingam — COMMENTED** [review]
Coverage has to be increased for HyperwalletAuthenticationToken file.

**Review by ramahalingam — APPROVED** [review]

File: src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java (hunk 1 above)
Comment (Rrathinasabapath) [reply src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java:74]: Done

File: src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java (hunk 2 above)
Comment (Rrathinasabapath) [reply src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java:160]: Done for the whole file

File: src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java (hunk 1 above)
Comment (Rrathinasabapath) [reply src/main/java/com/hyperwallet/clientsdk/util/HyperwalletApiClient.java]: Reverted the changes

File: src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java (hunk 2 above)
Comment (Rrathinasabapath) [reply src/main/java/com/hyperwallet/clientsdk/Hyperwallet.java]: Made changes for the whole file.

File: src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java (hunk 3 above)
Comment (Rrathinasabapath) [reply src/test/java/com/hyperwallet/clientsdk/HyperwalletIT.java]: enabled and tested the test case.

**Review by vkorotchenko — APPROVED** [review]
